### PR TITLE
Use SageMaker Containers MPI entry point

### PR DESCRIPTION
**Pull request by laurenyu**
This PR migrates the Chainer images to use the MPI support code we now have in SageMaker Containers. I opted to keep the same versions of mpi4py and Open MPI that were previously in the Chainer image rather than use the ones in the TF images to reduce the number of changes, but otherwise I tried to keep consistent with how TF uses the MPI support code.

This change fixes two things:
1. a problem where the image would hang on training jobs
1. a problem with Local Mode that was temporarily masked by #78 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://2hpflh3jba.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e0c6fe0c-7b87-4be6-aa86-1a1b92d50403%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://2hpflh3jba.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c155498e-2e99-41e8-a3f1-9f7ab3e1d681%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: test/resources/mnist/distributed_customer_script_with_env_vars.py [hunk 1]
```diff
@@ -162,12 +163,8 @@ def _preprocess_mnist(raw, withlabel, ndim, scale, image_dtype, label_dtype, rgb
 
     trainer.run()
 
-    # the MPI command makes the SM_HOSTS variable an invalid JSON
-    hosts = os.environ['SM_HOSTS']
-    master_node = hosts[1:hosts.find(',')]
-
     # only save the model in the master node
-    if args.host == master_node:
+    if args.host == json.loads(os.environ['SM_HOSTS'])[0]:
```
Comment (icywang86rui) [inline test/resources/mnist/distributed_customer_script_with_env_vars.py]: I would just put hosts in argparser as well. YMMV

Comment (laurenyu) [reply test/resources/mnist/distributed_customer_script_with_env_vars.py]: will do in a separate PR since this one's getting lengthy

**Pull request merged**